homeassistant/components/switch/xiaomi_miio.py

```diff
@@ -160,9 +181,117 @@ def async_update(self):
 
             self._state = state.is_on
             self._state_attrs.update({
+                ATTR_TEMPERATURE: state.temperature
+            })
+
+        except DeviceException as ex:
+            _LOGGER.error("Got exception while fetching the state: %s", ex)
+
+
+class XiaomiPowerStripSwitch(XiaomiPlugGenericSwitch, SwitchDevice):
+    """Representation of a Xiaomi Power Strip."""
+
+    # TODO: Add support for set_power_mode: normal/eco
+
+    def __init__(self, name, plug, device_info):
+        """Initialize the plug switch."""
+        XiaomiPlugGenericSwitch.__init__(self, name, plug, device_info)
+
+        self._state_attrs = {
+            ATTR_TEMPERATURE: None,
+            ATTR_LOAD_POWER: None,
+            ATTR_MODEL: self._device_info.raw['model'],
+        }
+
+    @asyncio.coroutine
+    def async_update(self):
+        """Fetch state from the device."""
+        from mirobo import DeviceException
+
+        # On state change the device doesn't provide the new state immediately.
+        if self._skip_update:
+            self._skip_update = False
+            return
+
+        try:
+            state = yield from self.hass.async_add_job(self._plug.status)
+            _LOGGER.debug("Got new state: %s", state)
+
+            self._state = state.is_on
+            self._state_attrs.update({
                 ATTR_TEMPERATURE: state.temperature,
-                ATTR_LOAD_POWER: state.load_power,
+                # FIXME: The device implementation of the power strip at
+                #        python-mirobo needs to be fixed.
```

Comment:
I think this fix should be merged first to `python-mirobo` first before merging this PR.

Reply:
I agree. It's already fixed and will be part of the first python-miio release.